CONTRIBUTING.rst

```diff
@@ -42,23 +42,55 @@ Contributing
 
 **Step 1**: Create an `issue <https://github.com/rucio/rucio/issues/new>`_ with the description
 of the contribution (motivation, modification and expected results).
-Every issue will get a **unique issue number**.
+Every issue will get a **unique issue number**. Valid component names are
+listed in the `label list <https://github.com/rucio/rucio/labels>`_.
 
 **Step 2**: Create a local branch that corresponds to the issue. There are utility scripts to help you with this::
 
-  $ ./tools/create-patch-branch <unique issue number> '<component> #<issue number>: <short_change_message>'
-  $ ./tools/create-feature-branch <unique issue number> '<component> #<issue number>: <short_change_message>'
+  $ ./tools/create-patch-branch <unique issue number> '<short_change_message>'
+  $ ./tools/create-feature-branch <unique issue number> '<short_change_message>'
 
 **Step 3**: Commit your change. The format of the commit message must be::
 
 <component>: <change_message> #<issue number>
 
-Valid component names are listed in the `label list <https://github.com/rucio/rucio/labels>`_
+Valid component names are listed in the `label list <https://github.com/rucio/rucio/labels>`_.
+
+If you add a `github-recognised keyword <https://help.github.com/articles/closing-issues-using-keywords/>`_ then
+the associated issue can be closed automatically once the pull request is merged, e.g.::
+
+    <component>: <change_message> Fix #<issue number>
 
 **Step 4**: Push the commit to your forked repository and create the pull request(s). There is a helper script to assist you::
 
   $ ./tools/submit-pull-request
 
+The helper script will propagate the commit message as the pull request title.
+
+If you use different tools to create pull requests like the `github interface <https://help.github.com/articles/creating-a-pull-request/>`_
+or the git command-line wrapper `hub <https://hub.github.com>`_, the following logic must be applied:
+ * If the contribution is a **patch**, the pull request must go against the **next** and the **master** branch.
```

Comment:
"must go" might be a bit confusing for some readers. I would write:

> If the contribution is a patch, two pull requests must be created, one for the next and one for the master branch. 

Same for the feature.